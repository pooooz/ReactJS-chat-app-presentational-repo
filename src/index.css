body {
  margin: 0;
  font-family: 'Press Start 2P', cursive;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
    background-color: black;
}

#root {
    display: flex;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}
