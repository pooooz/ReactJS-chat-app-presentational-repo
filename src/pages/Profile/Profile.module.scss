.profile {
  color: white;
  font-family: 'Press Start 2P', cursive;
}

.profile_header {
  text-align: center;
}

.profile_theme {
  display: flex;
  justify-content: space-evenly;
}

.profile_changeTheme {
  font-family: 'Press Start 2P', cursive !important;
}

.profile_name {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 25px 0 0 0;
}

.profile_changeName {
  display: flex;
  margin: 20px 0 0 0;
}

.profile_changeNameButton {
  font-family: 'Press Start 2P', cursive !important;
}

.profile_nameSection {
  display: flex;
  flex-direction: column;
  text-align: center;
}

.profile_visibility {
  min-width: 380px;
  display: flex;
  justify-content: space-between;
  margin: 20px 0 0 0;
}

.profile_visibilityValuePositive {
  font-family: 'Press Start 2P', cursive;
  background: transparent;
  border: none;
  outline: none;
  cursor: pointer;
  text-decoration: underline;
  color: #0bb70b;
}

.profile_visibilityValueNegative {
  font-family: 'Press Start 2P', cursive;
  background: transparent;
  border: none;
  outline: none;
  cursor: pointer;
  text-decoration: underline;
  color: #500419;
}