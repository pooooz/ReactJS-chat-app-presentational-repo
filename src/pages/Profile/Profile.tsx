import React, { FC, useContext, useState } from 'react';
import { ThemeContext } from 'src/utils/ThemeContext';
import { useDispatch, useSelector } from 'react-redux';

import { toggleProfile, changeName } from 'src/store/profile/actions';
import { selectName, selectVisible } from 'src/store/profile/selectors';

import { Button, ThemeProvider } from '@mui/material';
import { createTheme } from '@mui/material/styles';
import styles from './Profile.module.scss';
import { Input } from 'src/components/Input/Input';

const muiTheme = createTheme({
  typography: {
    fontFamily: 'Press Start 2P',
  },
});

export const Profile: FC = () => {
  const [value, setValue] = useState('');
  const { theme, toggleTheme } = useContext(ThemeContext);
  const dispatch = useDispatch();

  const visible = useSelector(selectVisible);
  const name = useSelector(selectName);

  const handleChangeName = () => {
    if (value) {
      dispatch(changeName(value));
    }
    setValue('');
  };

  return (
    <ThemeProvider theme={muiTheme}>
      <section className={styles.profile}>
        <h1 className={styles.profile_header}>Profile</h1>
        <div className={styles.profile_theme}>
          <p>{theme === 'light' ? 'ðŸŒž LIGHT' : 'ðŸŒ™ DARK'} </p>
          <Button
            onClick={toggleTheme}
            className={styles.profile_changeTheme}
            style={{
              width: '100px',
              color: '#32CD32FF',
            }}
          >
            Change theme
          </Button>
        </div>
        <section className={styles.profile_name}>
          <div className={styles.profile_nameSection}>
            <h3>Your name: {name}</h3>
            <div className={styles.profile_visibility}>
              Hide name in chat?
              <button
                className={
                  visible
                    ? styles.profile_visibilityValuePositive
                    : styles.profile_visibilityValueNegative
                }
                onClick={() => dispatch(toggleProfile())}
              >
                {visible ? 'Yes!' : 'No!'}
              </button>
            </div>
          </div>

          <div className={styles.profile_changeName}>
            <Input
              value={value}
              change={(event) => setValue(event.target.value)}
            />
            <Button
              onClick={handleChangeName}
              className={styles.profile_changeNameButton}
              style={{
                width: '100px',
                color: '#32CD32FF',
              }}
            >
              Change name
            </Button>
          </div>
        </section>
      </section>
    </ThemeProvider>
  );
};
