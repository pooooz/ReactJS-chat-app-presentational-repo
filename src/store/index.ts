import { createStore, compose, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';

import { profileReducer } from 'src/store/profile/reducer';
import { dialoguesReducer } from 'src/store/dialogues/reducer';

declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
  }
}

const persistConfig = {
  key: 'root',
  storage,
  blacklist: ['profile'],
};

export type StoreState = ReturnType<typeof rootReducer>;
const rootReducer = combineReducers({
  profile: profileReducer,
  dialogues: dialoguesReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const store = createStore(
  persistedReducer,
  composeEnhancers(applyMiddleware(thunk))
);

export const persistor = persistStore(store);
