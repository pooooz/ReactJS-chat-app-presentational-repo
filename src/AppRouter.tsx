import React, { FC } from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';

import { Sidebar } from './components/Sidebar/Sidebar';
import { Dialogue } from './pages/Dialogues/Dialogue';
import { ChatList } from './pages/Dialogues/components/ChatList/ChatList';
import { Profile } from 'src/pages/Profile/Profile';

export const AppRouter: FC = () => {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Sidebar />}>
          <Route path="profile" element={<Profile />} />
          <Route path="chats">
            <Route index element={<ChatList />} />
            <Route path=":chatId" element={<Dialogue />} />
          </Route>
        </Route>

        <Route path="*" element={<h2 style={{ color: '#00BFFF' }}>404</h2>} />
      </Routes>
    </BrowserRouter>
  );
};
