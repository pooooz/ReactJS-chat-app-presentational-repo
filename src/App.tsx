import React, { FC, useState } from 'react';
import { defaultContext, ThemeContext } from 'src/utils/ThemeContext';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';

import { AppRouter } from 'src/AppRouter';
import { store, persistor } from 'src/store';

export const App: FC = () => {
  const [theme, setTheme] = useState(defaultContext.theme);

  const toggleTheme = () => {
    setTheme(theme === 'light' ? 'dark' : 'light');
  };

  return (
    <Provider store={store}>
      <ThemeContext.Provider value={{ theme, toggleTheme }}>
        <PersistGate persistor={persistor}>
          <AppRouter />
        </PersistGate>
      </ThemeContext.Provider>
    </Provider>
  );
};
