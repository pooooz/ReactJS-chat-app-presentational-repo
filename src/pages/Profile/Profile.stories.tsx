import React, { useState } from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { Provider } from 'react-redux';

import { Profile } from './Profile';
import { store } from 'src/store';
import { defaultContext, ThemeContext } from 'src/utils/ThemeContext';

export default {
  title: 'pages/Profile/Profile',
  component: Profile,
} as ComponentMeta<typeof Profile>;

export const Default: ComponentStory<typeof Profile> = () => {
  const [theme, setTheme] = useState(defaultContext.theme);
  const toggleTheme = () => {
    setTheme(theme === 'light' ? 'dark' : 'light');
  };
  return (
    <Provider store={store}>
      <ThemeContext.Provider value={{ theme, toggleTheme }}>
        <Profile />
      </ThemeContext.Provider>
    </Provider>
  );
};
