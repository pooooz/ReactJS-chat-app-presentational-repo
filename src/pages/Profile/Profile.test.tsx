import React, { useState } from 'react';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import '@testing-library/user-event';

import { Profile } from './Profile';
import { store } from 'src/store';
import { defaultContext, ThemeContext } from 'src/utils/ThemeContext';

describe('Profile', () => {
  it('Render component', () => {
    render(
      <Provider store={store}>
        <Profile />
      </Provider>
    );
  });

  it('Toggle profile theme', () => {
    const ProfileWitThemeContext = () => {
      const [theme, setTheme] = useState(defaultContext.theme);
      const toggleTheme = () => {
        setTheme(theme === 'light' ? 'dark' : 'light');
      };
      return (
        <Provider store={store}>
          <ThemeContext.Provider value={{ theme, toggleTheme }}>
            <Profile />
          </ThemeContext.Provider>
        </Provider>
      );
    };
    render(<ProfileWitThemeContext />);
    fireEvent.click(screen.getAllByRole('button')[0]);
    expect(screen.getByText(/DARK/)).toBeInTheDocument();
  });

  it('Empty name value input', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Profile />
      </Provider>
    );
    fireEvent.input(screen.getByRole('textbox'), {
      target: { value: '' },
    });
    fireEvent.click(screen.getAllByRole('button')[2]);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Change name', async () => {
    render(
      <Provider store={store}>
        <Profile />
      </Provider>
    );
    fireEvent.input(screen.getByRole('textbox'), {
      target: { value: 'My name' },
    });
    fireEvent.click(screen.getAllByRole('button')[2]);
    await waitFor(() =>
      expect(screen.getByText(/My name/)).toBeInTheDocument()
    );
  });

  it('Toggle profile visibility', () => {
    render(
      <Provider store={store}>
        <Profile />
      </Provider>
    );
    fireEvent.click(screen.getAllByRole('button')[1]);
    expect(screen.getByText(/No!/)).toBeInTheDocument();
  });
});
