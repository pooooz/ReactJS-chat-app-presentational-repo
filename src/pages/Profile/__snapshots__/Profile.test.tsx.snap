// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Empty name value input 1`] = `
<DocumentFragment>
  <section
    class="profile"
  >
    <h1
      class="profile_header"
    >
      Profile
    </h1>
    <div
      class="profile_theme"
    >
      <p>
        ðŸŒž LIGHT 
      </p>
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root profile_changeTheme css-1t0zgcr-MuiButtonBase-root-MuiButton-root"
        style="width: 100px; color: rgba(50, 205, 50, 1);"
        tabindex="0"
        type="button"
      >
        Change theme
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <section
      class="profile_name"
    >
      <div
        class="profile_nameSection"
      >
        <h3>
          Your name: Anonymous
        </h3>
        <div
          class="profile_visibility"
        >
          Hide name in chat?
          <button
            class="profile_visibilityValuePositive"
          >
            Yes!
          </button>
        </div>
      </div>
      <div
        class="profile_changeName"
      >
        <div
          class="input_wrap"
        >
          <input
            class="input_green"
            type="text"
            value=""
          />
        </div>
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root profile_changeNameButton css-1t0zgcr-MuiButtonBase-root-MuiButton-root"
          style="width: 100px; color: rgba(50, 205, 50, 1);"
          tabindex="0"
          type="button"
        >
          Change name
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </section>
  </section>
</DocumentFragment>
`;
